### PR TITLE
Fix gitlab and bitbucket branch comparison URLs

### PR DESCRIPTION
**Pull request by mtsgrd**
- we don't want to provide fork information to forges unless actually forked

### CODE REVIEW

File: apps/desktop/src/lib/baseBranch/baseBranchService.ts [hunk 1]
```diff
@@ -29,7 +29,10 @@ export class BaseBranchService {
 	// Deduplicated push repo information.
 	readonly pushRepo = shallowDeduplicate(
 		derived(this.base, (base) => {
-			return base ? parseRemoteUrl(base.pushRemoteUrl) : undefined;
+			// TODO: We should not send `pushRemoteUrl` if it's the same as `remoteUrl`.
+			if (base?.pushRemoteUrl && base.pushRemoteUrl !== base.remoteUrl) {
+				return parseRemoteUrl(base.pushRemoteUrl);
```
Comment (estib-vega) [inline apps/desktop/src/lib/baseBranch/baseBranchService.ts]: Is the comment still needed?
It seems like the condition already is taking care of what your TODO describes

**Review by estib-vega — APPROVED** [review]
Only that question about the comment

File: apps/desktop/src/lib/baseBranch/baseBranchService.ts (hunk 1 above)
Comment (mtsgrd) [reply apps/desktop/src/lib/baseBranch/baseBranchService.ts]: The comment is about the back end sending identical remoteUrl and pushRemoteUrl. I think we should only send pushRemoteUrl if it's different from remoteUrl, but I also think we should kick that can down the road.

Comment (mtsgrd) [reply apps/desktop/src/lib/baseBranch/baseBranchService.ts]: Let me clarify the comment!

Comment (estib-vega) [reply apps/desktop/src/lib/baseBranch/baseBranchService.ts]: Gotcha 👍

**Pull request merged**